pkg/trustedresources/verify.go

```diff
@@ -157,13 +211,47 @@ func prepareObjectMeta(in metav1.ObjectMeta) (metav1.ObjectMeta, []byte, error)
 	return out, signature, nil
 }
 
-// getVerifiers get all verifiers from configmap
-func getVerifiers(ctx context.Context, k8s kubernetes.Interface) ([]signature.Verifier, error) {
-	cfg := config.FromContextOrDefaults(ctx)
+// getVerifiersFromPolicy get all verifiers from configmap
+func getVerifiersFromPolicy(ctx context.Context, k8s kubernetes.Interface, source string, verificationpolicy *v1alpha1.VerificationPolicy) ([]signature.Verifier, error) {
 	verifiers := []signature.Verifier{}
-	// TODO(#5527): consider using k8s://namespace/name instead of mounting files.
+	for _, a := range verificationpolicy.Spec.Authorities {
+		algorithm, err := hashAlgorithm(a.Key.HashAlgorithm)
+		if err != nil {
+			continue
+		}
+		if len(a.Key.Data) != 0 {
+			for _, d := range a.Key.Data {
+				publicKey, err := cryptoutils.UnmarshalPEMToPublicKey([]byte(d))
+				if err != nil || publicKey == nil {
+					return nil, fmt.Errorf("authority %q contains an invalid public key: %w", a.Name, err)
+				}
+				verifier, err := signature.LoadVerifier(publicKey, algorithm)
+				if err == nil {
+					verifiers = append(verifiers, verifier)
+				}
+			}
+		} else if a.Key.SecretRef != nil {
+			verifiersFromSecret, keysData, err := verifierForKeyRef(ctx, fmt.Sprintf("%s%s/%s", keyReference, a.Key.SecretRef.Namespace, a.Key.SecretRef.Name), algorithm, k8s)
+			if err == nil {
+				verifiers = append(verifiers, verifiersFromSecret...)
+				a.Key.Data = append(a.Key.Data, keysData...)
```

Comment:
It'd be great to add a comment explaining what this is doing, especially since we're mutating local state.

(this is part of the reason why policy-controller caches data in a config map - it's dumping it to a separate location so that the original struct isn't modified).

Reply:
alternatively you could just store this data in a separate cache.

Reply:
Yes this is a great point!
How about we removing this in this PR? I think policy-controller's internal conversion is a good way. Maybe that would be something we can follow/consider in future PRs

Reply:
This is removed in current PR, I think configmap conversion would be a good way to consider